File: backend/app/schemas/ai.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


class ErrorAnalysisContext(BaseModel):
    error_id: int
    programming_language: str
    framework: str | None
    error_message: str
    stack_trace: str
    file: str | None
    line: int | None
    code_snippet: str | None
    command: str
    project_name: str


class AIAnalysisContent(BaseModel):
    cause: str
    explanation: str
    solution_steps: list[str]
    code_fix: str | None
    terminal_commands: list[str]


class AIAnalyzeRequest(BaseModel):
    confirmed: Literal[True]


class AIAnalysisRead(AIAnalysisContent):
    model_config = ConfigDict(from_attributes=True)

    id: int
    error_id: int
    provider: str
    model: str
    created_at: datetime


class AIStatusResponse(BaseModel):
    provider: Literal["gemini"] = "gemini"
    configured: bool
    model: str


class JournalCommit(BaseModel):
    commit: str
    message: str
    files: list[str]


class JournalCommand(BaseModel):
    name: str
    command: str
    status: str


class JournalError(BaseModel):
    message: str
    file: str | None
    line: int | None
    ai_analyzed: bool
    ai_resolution: str | None = None


class DevelopmentJournalContext(BaseModel):
    project_id: int
    project_name: str
    date: str
    branch: str | None
    commits: list[JournalCommit]
    changed_files: list[str]
    commands: list[JournalCommand]
    errors: list[JournalError]


class AIJournalContent(BaseModel):
    title: str = Field(min_length=1, max_length=200)
    content: str = Field(min_length=1, max_length=100_000)
    tags: list[str] = Field(default_factory=list, max_length=20)


class AIJournalGenerateRequest(BaseModel):
    confirmed: Literal[True]


class AIJournalRead(AIJournalContent):
    model_config = ConfigDict(from_attributes=True)

    id: int
    project_id: int
    project_name: str
    provider: str
    model: str
    source_counts: dict[str, int]
    created_at: datetime

File: frontend/src/types/ai.ts
export interface AIStatus {
  provider: 'gemini'
  configured: boolean
  model: string
}

export interface ErrorAnalysisContext {
  error_id: number
  programming_language: string
  framework: string | null
  error_message: string
  stack_trace: string
  file: string | null
  line: number | null
  code_snippet: string | null
  command: string
  project_name: string
}

export interface AIAnalysis {
  id: number
  error_id: number
  provider: string
  model: string
  cause: string
  explanation: string
  solution_steps: string[]
  code_fix: string | null
  terminal_commands: string[]
  created_at: string
}

export interface JournalCommit {
  commit: string
  message: string
  files: string[]
}

export interface JournalCommand {
  name: string
  command: string
  status: string
}

export interface JournalError {
  message: string
  file: string | null
  line: number | null
  ai_analyzed: boolean
  ai_resolution: string | null
}

export interface DevelopmentJournalContext {
  project_id: number
  project_name: string
  date: string
  branch: string | null
  commits: JournalCommit[]
  changed_files: string[]
  commands: JournalCommand[]
  errors: JournalError[]
}

export interface AIJournalDraft {
  id: number
  project_id: number
  project_name: string
  provider: string
  model: string
  title: string
  content: string
  tags: string[]
  source_counts: Record<string, number>
  created_at: string
}
